cli/internal/helm/versionlister.go

```diff
@@ -0,0 +1,141 @@
+}
+
+// listAction execute a List action by wrapping helm's action package.
+// It creates the action, runs it at returns results and errors.
+func (c ReleaseVersionClient) listAction(release string) (res []*release.Release, err error) {
+	action := action.NewList(c.config)
+	action.Filter = release
+	// during init, the kube API might not yet be reachable, so we retry
+	err = retry.OnError(retry.DefaultBackoff, func(err error) bool {
+		return err != nil
+	}, func() error {
+		res, err = action.Run()
+		return err
+	})
+	return
+}
+
+// Versions queries the cluster for running versions and returns a map of releaseName -> version.
+func (c ReleaseVersionClient) Versions() (ServiceVersions, error) {
+	ciliumVersion, err := c.currentVersion(ciliumInfo.releaseName)
+	if err != nil {
+		return ServiceVersions{}, fmt.Errorf("getting %s version: %w", ciliumInfo.releaseName, err)
+	}
+	certManagerVersion, err := c.currentVersion(certManagerInfo.releaseName)
+	if err != nil {
+		return ServiceVersions{}, fmt.Errorf("getting %s version: %w", certManagerInfo.releaseName, err)
+	}
+	operatorsVersion, err := c.currentVersion(constellationOperatorsInfo.releaseName)
+	if err != nil {
+		return ServiceVersions{}, fmt.Errorf("getting %s version: %w", constellationOperatorsInfo.releaseName, err)
+	}
+	servicesVersion, err := c.currentVersion(constellationServicesInfo.releaseName)
+	if err != nil {
+		return ServiceVersions{}, fmt.Errorf("getting %s version: %w", constellationServicesInfo.releaseName, err)
+	}
+	csiVersions, err := c.csiVersions()
+	if err != nil {
+		return ServiceVersions{}, fmt.Errorf("getting CSI versions: %w", err)
+	}
+
+	serviceVersions := ServiceVersions{
+		cilium:                 ciliumVersion,
+		certManager:            certManagerVersion,
+		constellationOperators: operatorsVersion,
+		constellationServices:  servicesVersion,
+		csiVersions:            csiVersions,
+	}
+
+	if awsLBVersion, err := c.currentVersion(awsLBControllerInfo.releaseName); err == nil {
+		serviceVersions.awsLBController = awsLBVersion
+	} else if !errors.Is(err, errReleaseNotFound) {
+		return ServiceVersions{}, fmt.Errorf("getting %s version: %w", awsLBControllerInfo.releaseName, err)
+	}
+
+	return serviceVersions, nil
+}
+
+// currentVersion returns the version of the currently installed helm release.
+func (c ReleaseVersionClient) currentVersion(release string) (semver.Semver, error) {
+	rel, err := c.listAction(release)
+	if err != nil {
+		return semver.Semver{}, err
+	}
+
+	if len(rel) == 0 {
+		return semver.Semver{}, errReleaseNotFound
+	}
+	if len(rel) > 1 {
+		return semver.Semver{}, fmt.Errorf("multiple releases found for %s", release)
+	}
+
+	if rel[0] == nil || rel[0].Chart == nil || rel[0].Chart.Metadata == nil {
+		return semver.Semver{}, fmt.Errorf("received invalid release %s", release)
+	}
+	return semver.New(rel[0].Chart.Metadata.Version)
+}
+
+// csi versions needs special handling because all versions of its subcharts should be gathered.
```

Comment:
```suggestion
// csi versions needs special handling because all versions of its subcharts should be gathered.
// If the CSI chart is not installed, no error is returned because the user can configure if the chart should be installed.
```